sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -187,6 +187,10 @@ private SentryEvent processEvent(
   public void captureUserFeedback(UserFeedback userFeedback) {
     Objects.requireNonNull(userFeedback, "SentryEvent is required.");
 
+    if (userFeedback.getEventId() == null || userFeedback.getEventId() == SentryId.EMPTY_ID) {
```

Comment:
As far as I can see `userFeedback#eventId` can never be `null`, thus should be marked as `@NotNull` and this part of if statement wouldn't be needed.

Reply:
doesn't this assume tooling respects the annotations? aren't those just hints that tooling must support? Or it's guaranteed compiler won't let this compile?

Reply:
those annotations are just a convention adopted by IntelliJ based IDEs, so it still compiles (for Java-even with warnings), but for Kotlin, it doesn't compile iirc.
I think its fine to keep this `if` for the sake of safety, but annotating as @maciejwalkowiak said is also helpful

Reply:
Not respecting this annotation should result in warning that we must fix. Not sure if currently the build fails but I believe it should.